### ISSUE TITLE
BLD: `fft`: c++: error "Invalid argument" in `pypocketfft` on Windows jobs

### ISSUE THREAD

**Issue by mdhaber** [scipy/scipy#21385]
### Describe your issue.

Windows jobs (e.g. PR gh-21339, gh-21277) on CI are failing with the following message:

### Reproducing Code Example

```python
-
```

### Error message

```shell
2024-08-13T20:59:58.1193394Z [735/1283] Compiling C object scipy/stats/_rcont/rcont.cp310-win_amd64.pyd.p/.._libnpyrandom_distributions.c.obj
2024-08-13T20:59:58.1194297Z [736/1283] Compiling C++ object scipy/fft/_pocketfft/pypocketfft.cp310-win_amd64.pyd.p/pypocketfft.cxx.obj
2024-08-13T20:59:58.1195078Z FAILED: scipy/fft/_pocketfft/pypocketfft.cp310-win_amd64.pyd.p/pypocketfft.cxx.obj 
2024-08-13T20:59:58.1199380Z "c++" "-Iscipy\fft\_pocketfft\pypocketfft.cp310-win_amd64.pyd.p" "-Iscipy\fft\_pocketfft" "-I..\scipy\fft\_pocketfft" "-I..\scipy\_lib\pocketfft" "-IC:\hostedtoolcache\windows\Python\3.10.11\x64\Include" "-fvisibility=hidden" "-fvisibility-inlines-hidden" "-fdiagnostics-color=always" "-D_GLIBCXX_ASSERTIONS=1" "-D_FILE_OFFSET_BITS=64" "-Wall" "-Winvalid-pch" "-std=c++17" "-O2" "-g" "-D__USE_MINGW_ANSI_STDIO=1" "-DMS_WIN64=" "'-IC:\hostedtoolcache\windows\Python\3.10.11\x64\Include'" "'-IC:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\pybind11\include'" "-DPOCKETFFT_NO_MULTITHREADING" "-DPOCKETFFT_CACHE_SIZE=16" -MD -MQ scipy/fft/_pocketfft/pypocketfft.cp310-win_amd64.pyd.p/pypocketfft.cxx.obj -MF "scipy\fft\_pocketfft\pypocketfft.cp310-win_amd64.pyd.p\pypocketfft.cxx.obj.d" -o scipy/fft/_pocketfft/pypocketfft.cp310-win_amd64.pyd.p/pypocketfft.cxx.obj "-c" ../scipy/fft/_pocketfft/pypocketfft.cxx
2024-08-13T20:59:58.6263380Z c++: error: '-IC:\hostedtoolcache\windows\Python\3.10.11\x64\Include': Invalid argument
2024-08-13T20:59:58.6265218Z c++: error: '-IC:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\pybind11\include': Invalid argument
2024-08-13T20:59:58.6266610Z [737/1283] Generating 'scipy\\stats\\_rcont\\rcont.cp310-win_amd64.pyd.p\\rcont.c'
2024-08-13T20:59:58.6267467Z ninja: build stopped: subcommand failed.
2024-08-13T20:59:58.6375398Z Build failed!
2024-08-13T20:59:58.7881965Z ##[error]Process completed with exit code 1.
```

### SciPy/NumPy/Python version and system information

```shell
-
```

**Comment by adammj** [comment]
Yeah, I was wondering what that error is, since it bears no relation to the code that was changed.

**Comment by WarrenWeckesser** [comment]
Two of the arguments being passed to the compiler are `"'-IC:\hostedtoolcache\windows\Python\3.10.11\x64\Include'"` and  `"'-IC:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\pybind11\include'"`.  Note the single quotes inside the double quotes.  Is that normal on Windows?  None of the other arguments have those extra single quotes.

**Comment by WarrenWeckesser** [comment]
This pybind11 issue looks like our issue: https://github.com/pybind/pybind11/issues/5300

**Comment by WarrenWeckesser** [comment]
This is fixed by pybind11 2.13.4, which was just released.
